# Observing strategy for Nov 29, Nov 30 2024

Prepare the observations (finding charts, visibility charts, design observing strategy,) for two observing nights at AS Vidojevica, 29.11 and 30.11.2024, for 1.4m Milankovic telescope.

In [1]:
from os import listdir
from astroquery.simbad import Simbad
import numpy as np
from astropy.io import fits

import matplotlib.pyplot as plt
plt.style.use('seaborn-v0_8')
# plt.style.use('seaborn-v0_8-colorblind')

from astropy.coordinates import SkyCoord
import astropy.units as u

import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

## Gathering Basic Information

Prepare object log table, with basic information: name, object type, visual magnitude, coordinates from the SIMBAD database (e.g. for standard epoch J2000.0, test SIMBAD search options to get coordinates for the epoch of observation).

In [2]:
targets = ['SNR G130.7+03.1',
           'Mrk 335', 'IRAS 03450+0055', 'Mrk 1506', 'IC 450', 'Mrk 79', 'Mrk 110','NGC 3516','Mrk 279', 'NGC 3227', 'NGC 4051', 'NGC 4151', 'Mrk 817']

user_targets = ['SNR G130.7+03.1',
           'Mrk 335', 'IRAS 03450+0055', 'Mrk 1506', 'IC 450', 'Mrk 79', 'Mrk 110','NGC 3516','Mrk 279', 'NGC 3227', 'NGC 4051', 'NGC 4151', 'Mrk 817']

user_targets[0] = 'Pa30'
user_targets[2] = 'IRAS03450'

In [3]:
### Looking at the simbad columns available to us
simbad_table_rows = Simbad.list_votable_fields()[["name", "description"]]
simbad_table_rows.to_pandas()

name                                        description
0            mesDiameter                   Collection of stellar diameters.
1                  mesPM                      Collection of proper motions.
2                 mesISO    Infrared Space Observatory (ISO) observing log.
3                 mesSpT                      Collection of spectral types.
4              allfluxes   all flux/magnitudes U,B,V,I,J,H,K,u_,g_,r_,i_,z_
5                  ident              Identifiers of an astronomical object
6                   flux  Magnitude/Flux information about an astronomic...
7               mesOtype  Other object types associated with an object w...
8                 mesPLX            Collection of trigonometric parallaxes.
9               otypedef     all names and definitions for the object types
10           mesDistance  Collection of distances (pc, kpc or Mpc) by se...
11                otypes  List of all object types associated with an ob...
12                mesVar  Collection of stellar variability types and pe...
13                mesXmm                                 XMM observing log.
14         mesVelocities         Collection of HRV, Vlsr, cz and redshifts.
15               has_ref  Associations between astronomical objects and ...
16                mesRot                     Stellar Rotational Velocities.
17                biblio                                       Bibliography
18                   ids                   all names concatenated with pipe
19           mesHerschel                         The Herschel observing Log
20                mesIUE  International Ultraviolet Explorer observing log.
21               mesFe_h  Collection of metallicity, as well as Teff, lo...
22              alltypes            all object types concatenated with pipe
23                 basic          General data about an astronomical object
24                   dec                                        Declination
25               main_id                      Main identifier for an object
26             otype_txt                                        Object type
27                    ra                                    Right ascension
28           coo_bibcode                               Coordinate reference
29         coo_err_angle                             Coordinate error angle
30           coo_err_maj                        Coordinate error major axis
31      coo_err_maj_prec              Coordinate error major axis precision
32           coo_err_min                        Coordinate error minor axis
33      coo_err_min_prec              Coordinate error minor axis precision
34              coo_qual                                 Coordinate quality
35        coo_wavelength  Wavelength class for the origin of the coordin...
36              dec_prec                              Declination precision
37          galdim_angle                               Galaxy ellipse angle
38        galdim_bibcode                         Galaxy dimension reference
39        galdim_majaxis                            Angular size major axis
40   galdim_majaxis_prec                  Angular size major axis precision
41        galdim_minaxis                            Angular size minor axis
42   galdim_minaxis_prec                  Angular size minor axis precision
43           galdim_qual                           Galaxy dimension quality
44     galdim_wavelength  Wavelength class for the origin of the Galaxy ...
45                   hpx                         Healpix number at ORDER=10
46         morph_bibcode                       morphological type reference
47            morph_qual                         Morphological type quality
48            morph_type                                 Morphological type
49                 nbref                               number of references
50                   oid                         Object internal identifier
51                 otype                                        Obje

In [4]:
simbad = Simbad()
simbad.add_votable_fields("allfluxes", "B", "coordinates", "otypedef")
result_table = simbad.query_objects(targets)
result_table

<Table length=13>
    main_id             ra         ... user_specified_id object_number_id
                       deg         ...                                   
     object          float64       ...       object           int64      
--------------- ------------------ ... ----------------- ----------------
SNR G130.7+03.1            31.4043 ...   SNR G130.7+03.1                1
       Mrk  335 1.5814051412599996 ...   Mrk 335                        2
IRAS 03450+0055  56.91745322117999 ...   IRAS 03450+0055                3
       Mrk 1506     68.29623197304 ...   Mrk 1506                       4
        IC  450    103.05139427733 ...   IC 450                         5
       Mrk   79 115.63675444617002 ...   Mrk 79                         6
       Mrk  110    141.30353294399 ...   Mrk 110                        7
      NGC  3516      166.697763417 ...   NGC 3516                       8
       Mrk  279 208.26431206837998 ...   Mrk 279                        9
      NGC  3227    155.87740214554 ...   NGC 3227                      10
      NGC  4051 180.79004222387996 ...   NGC 4051                      11
      NGC  4151 182.63573325577997 ...   NGC 4151                      12
       Mrk  817    219.09200858818 ...   Mrk 817                       13

In [5]:
targets_basics =  result_table.to_pandas()[["main_id", "ra", "dec", "coo_err_angle", "otypedef.description", "G", "B", "V"]]
targets_basics['user_id'] = user_targets
targets_basics

main_id          ra        dec  coo_err_angle  \
0   SNR G130.7+03.1   31.404300  64.828300              0   
1          Mrk  335    1.581405  20.202949             90   
2   IRAS 03450+0055   56.917453   1.087236             90   
3          Mrk 1506   68.296232   5.354339             90   
4           IC  450  103.051394  74.426978             90   
5          Mrk   79  115.636754  49.809662             90   
6          Mrk  110  141.303533  52.286218             90   
7         NGC  3516  166.697763  72.568694             90   
8          Mrk  279  208.264312  69.308170             90   
9         NGC  3227  155.877402  19.865078             90   
10        NGC  4051  180.790042  44.531301             90   
11        NGC  4151  182.635733  39.405851             90   
12         Mrk  817  219.092009  58.794276             90   

   otypedef.description        G      B      V    user_id  
0     SuperNova Remnant      NaN    NaN    NaN       Pa30  
1      Seyfert 1 Galaxy  14.4294  14.19  13.85    Mrk 335  
2      Seyfert 1 Galaxy  14.4675  15.09  14.64  IRAS03450  
3      Seyfert 1 Galaxy  14.5595  15.72  15.05   Mrk 1506  
4      Seyfert 1 Galaxy  13.4548  15.16  14.19     IC 450  
5      Seyfert 1 Galaxy  15.3720  14.74  14.27     Mrk 79  
6      Seyfert 1 Galaxy  14.7290  16.82  16.41    Mrk 110  
7      Seyfert 1 Galaxy  15.3586  13.12  12.40   NGC 3516  
8      Seyfert 1 Galaxy  15.6056  15.15  14.46    Mrk 279  
9      Seyfert 1 Galaxy  14.0456  12.61  11.79   NGC 3227  
10     Seyfert 1 Galaxy  14.7241  11.08  12.92   NGC 4051  
11     Seyfert 1 Galaxy  13.3786  12.18  11.48   NGC 4151  
12     Seyfert 1 Galaxy  14.3838  14.19  13.79    Mrk 817

## Visibility Plots
Check the visibility of all objects for the selected dates and the location of AS Vidojevica using the e.g. Staralt program (or Airmass)

In [6]:
# Printing out names, RAs and DECs to input in staralt

for i, row in targets_basics.iterrows():
    user_id = row.user_id.replace(' ', '_')
    print(f'{user_id} {row.ra:07.3f} {row.dec:06.3f}')

Pa30 031.404 64.828
Mrk_335 001.581 20.203
IRAS03450 056.917 01.087
Mrk_1506 068.296 05.354
IC_450 103.051 74.427
Mrk_79 115.637 49.810
Mrk_110 141.304 52.286
NGC_3516 166.698 72.569
Mrk_279 208.264 69.308
NGC_3227 155.877 19.865
NGC_4051 180.790 44.531
NGC_4151 182.636 39.406
Mrk_817 219.092 58.794


For staralt, we obtain the following visibility plots for our 13 objects for the nights of the 29th and 30th November 2024.

<div style="display: flex; justify-content: space-around;">
  <img src="staralt_29-11-2025.png" width="600"/>
  <img src="staralt_30-11-2025.png" width="600"/>
</div>



In [7]:
# Printing out names to input in airmass

for i, row in targets_basics.iterrows():
    print(row.main_id)

SNR G130.7+03.1
Mrk  335
IRAS 03450+0055
Mrk 1506
IC  450
Mrk   79
Mrk  110
NGC  3516
Mrk  279
NGC  3227
NGC  4051
NGC  4151
Mrk  817



For airmass.org, we obtain the following visibility plots for our 13 objects for the nights of the 29th and 30th November 2024.
. The plot times are shown in UTC, and it should be noted that Serbian time is CET (UTC + 1). We also shade elevations $< 30$ and $> 70$, considering this elevation range to be optimal.

<div style="display: flex; justify-content: space-around;">
  <img src="airmass_29-11-2025.png" width="600"/>
  <img src="airmass_30-11-2025.png" width="600"/>
</div>

We also obtain some auxiliary data, which is as follows:

<div style="display: flex; justify-content: space-around;">
  <img src="airmass_data_29-11-2025.png" width="500"/>
  <img src="airmass_data_30-11-2025.png" width="500"/>
</div>

Lookig at airmall plots, we can potentially observe all of our objects, given that the exposures required aren't too long.

# Observation Plan

Judging by charts on the 29th and 30th, we can pretty much keep the same schedule on both days. Through the gift of foresight, we know that we need exposures on the order of seconds. Hence, we'll hopefully be able to take multiple exposures in all require bands withing a general 30-minute window per object. That being said, the schedule is a bit tight, and we should plan to abandon MRK 817 and MRK 279 in that order in case there are delays.

| Time (CET) | Observation                         |
| ---------- | ----------------------------------- |
| 1700-1745  | Bias, Flats in B, V, R, Ha, \[SII\] |
| 1600       | Darks                               |
| 1900       | MRK 335                             |
| 2100       | Pa30                                |
| 2200       | IRAS 03450                          |
| 2330       | MRK 1506                            |
| 2400       | MRK 79                              |
| 2430       | IC 450                              |
| 0100       | MRK 110                             |
| 0130       | NGC 3227                            |
| 0200       | NGC 4051                            |
| 0230       | NGC 4151                            |
| 0300       | NGC 3516                            |
| 0330       | MRK 279                             |
| 0400       | MRK 817                             |

## Finding Charts

We will use SAO DS9 to load up images of our objects, query SIMBAD to find stars not marked as variable from within DS9, and mark them. We'll need to figure out the field of view of the telescope. We are using the 1.4 m Milankovic telescope with the Andor iKon-L 936 CCD on port 1

The telescope has specifications:
- 11.2 m focal length
- f/8

The CCD has the following specifications:
- 0.65x focal reducer (resulting focal length is 7132 mm)
- f/7.6 field flattener (resulting in focal length 10.64 m)
- $13 \times 13~\mu m$ pixel size
- $2048 \times 2048$ resolution


We know that the pixel scale in this setup is then:
$$\frac{206.265 \times 13 ~ \mu m}{10640 ~ mm} = 0.26'' ~ pixel^{-1}$$

Hence, the field of view in arcmin is:
$$\frac{0.26'' ~ pixel^{-1} \times 2048}{60} \times \frac{0.26'' ~ pixel^{-1} \times 2048}{60} = 9 \times 9 ~ arcmin$$

Let's now list the finding charts for all of our objects. We will mark the object in a red box, and mark the stars in the FOV with green circles. The obejcts are ordered in order of the observation schedule.

<div style="display: flex; justify-content: space-around;">
  <img src="mrk335.png" width="300"/>
  <img src="pa30.png" width="300"/>
  <img src="iras03450.png" width="300"/>
  <img src="mrk1506.png" width="300"/>
</div>

<div style="display: flex; justify-content: space-around;">
  <img src="mrk79.png" width="300"/>
  <img src="ic450.png" width="300"/>
  <img src="mrk110.png" width="300"/>
  <img src="ngc3227.png" width="300"/>
</div>

<div style="display: flex; justify-content: space-around;">
  <img src="ngc4051.png" width="300"/>
  <img src="ngc4151.png" width="300"/>
  <img src="ngc3516.png" width="300"/>
  <img src="mrk279.png" width="300"/>
</div>

<img src="mrk817.png" width="300"/>

We'll collect necessary details for exposure time calculation.
- Seeing: 0.7"
- Radius for Photometry = $2 \times 0.7" = 1.4"$
- Total transparency = 0.8
- Sky magnitude = 19
- Desired SNR = 100


| UTC  | main_id    | point_source / extended | airmass | V_mag |
| ---- | ---------- | ----------------------- | ------- | ----- |
| 1900 | MRK 335    | ps                      | 1.08    | 13.85 |
| 2100 | Pa30       | ps                      | 1.09    | NaN   |
| 2200 | IRAS 03450 | ps                      | 1.37    | 14.64 |
| 2330 | MRK 1506   | ps                      | 1.26    | 15.05 |
| 2400 | MRK 79     | e                       | 1.13    | 14.27 |
| 2430 | IC 450     | ps                      | 1.19    | 14.19 |
| 0100 | MRK 110    | ps                      | 1.22    | 16.41 |
| 0130 | NGC 3227   | e                       | 1.67    | 11.79 |
| 0200 | NGC 4051   | e                       | 1.63    | 12.92 |
| 0230 | NGC 4151   | e                       | 1.60    | 11.48 |
| 0300 | NGC 3516   | e                       | 1.24    | 12.40 |
| 0330 | MRK 279    | ps                      | 1.45    | 14.46 |
| 0400 | MRK 817    | ps                      | 1.59    | 13.79 |

We'll calculate the exposure time for the brightest and the dimmest objects.

For NGC 4151:

<img src="ngc4151_exposure_time.png" width="700"/>



For MRK 110:

<img src="mrk110_exposure_time.png" width="700"/>


## Telescope Input Table

In [8]:
sexagesimal_ras = []
sexagesimal_decs = []

for i, row in targets_basics.iterrows():
    coord = SkyCoord(ra=row.ra * u.deg, dec=row.dec * u.deg, frame='icrs')
    ra_str = coord.ra.to_string(unit=u.hourangle, sep=':', precision=2, pad=True)
    dec_str = coord.dec.to_string(unit=u.deg, sep=':', precision=2, alwayssign=True, pad=True)
    sexagesimal_ras.append(ra_str)
    sexagesimal_decs.append(dec_str)

In [9]:
input_table = targets_basics[['main_id', 'ra', 'dec']]
input_table['ra'] = sexagesimal_ras
input_table['dec'] = sexagesimal_decs
input_table.rename(columns = {'main_id': 'name'})
input_table


/tmp/ipykernel_21672/3056933022.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  input_table['ra'] = sexagesimal_ras
/tmp/ipykernel_21672/3056933022.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  input_table['dec'] = sexagesimal_decs


main_id           ra           dec
0   SNR G130.7+03.1  02:05:37.03  +64:49:41.88
1          Mrk  335  00:06:19.54  +20:12:10.62
2   IRAS 03450+0055  03:47:40.19  +01:05:14.05
3          Mrk 1506  04:33:11.10  +05:21:15.62
4           IC  450  06:52:12.33  +74:25:37.12
5          Mrk   79  07:42:32.82  +49:48:34.78
6          Mrk  110  09:25:12.85  +52:17:10.39
7         NGC  3516  11:06:47.46  +72:34:07.30
8          Mrk  279  13:53:03.43  +69:18:29.41
9         NGC  3227  10:23:30.58  +19:51:54.28
10        NGC  4051  12:03:09.61  +44:31:52.68
11        NGC  4151  12:10:32.58  +39:24:21.06
12         Mrk  817  14:36:22.08  +58:47:39.39